tests/unit/test_exporter.py

```diff
@@ -105,7 +106,7 @@ def test_validate_config():
 def test_apply_config_success(mocker):
     """Test successfully applying snap configuration."""
     mock_stop = mocker.patch.object(exporter.ExporterSnap, "stop")
-    mock_start = mocker.patch.object(exporter.ExporterSnap, "start")
+    mock_start = mocker.patch.object(exporter.ExporterSnap, "restart")
```

Comment:
Why we change "start" to "restart" here?

Reply:
I think it's due this [line](https://github.com/canonical/charm-prometheus-juju-exporter/pull/32/files#diff-6c3b99b0837808d8b20d6867ec8d6948125b6764965f2b309e864453e2ae6ef9R182). But it's good question why we use start before and now restart.

Reply:
The reason is that, "start"-ing an already started snap daemon service would not apply the modified config, whereas "restart"-ing a snap will (and "restart"-ing an inactive snap have the same effect as "start"-ing it). Which means, with the previous implementation, if we change a config option after the snap service started, it will not take in the latest values. This is a bug I observed during my tests so I made the change here.

Reply:
I have just re-tested with `self.start()` and it indeed worked. I don't have the previous logs so not sure why I was not able to do live config change earlier. My assumption was somehow the snap didn't actually stop so `start()` couldn't make a difference on the running service. I would say we keep the change to `restart()` just to avoid the possibility of such a case.

Reply:
To me it's more logical to run `restart`, without any further information/knowledge.